src/uct/api/uct_def.h

```diff
@@ -223,6 +223,20 @@ typedef void (*uct_completion_callback_t)(uct_completion_t *self,
  */
 typedef ucs_status_t (*uct_pending_callback_t)(uct_pending_req_t *self);
 
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Callback to process peer failure.
+ *
+ * @param [in]  arg      User argument to be passed to the callback.
+ * @param [in]  ep       Endpoint to handle peer failure error,
+ *                       @a ep becomes unusable. All following operation on @a ep
+ *                       will return @a status error.
```

Comment:
I suggest the following rewording of the description for the ep argument:
  Endpoint which has failed.  Upon return from the callback, this @a ep is no longer usable and all subsequent operations on this @a ep will fail with the error code passed in @a status